barretenberg/cpp/src/barretenberg/protogalaxy/protogalaxy_prover.hpp

```diff
@@ -271,20 +292,31 @@ template <class ProverInstances_> class ProtoGalaxyProver_ {
      * polynomial, extract the value at row_idx. Use these values to create a univariate polynomial, and then extend
      * (i.e., compute additional evaluations at adjacent domain values) as needed.
      * @todo TODO(https://github.com/AztecProtocol/barretenberg/issues/751) Optimize memory
+     *
+     *
      */
-    void extend_univariates(ExtendedUnivariates& extended_univariates,
+
+    template <typename ExtendedUnivariatesType, size_t skip_count = 0>
+    void extend_univariates(ExtendedUnivariatesType& extended_univariates,
                             const ProverInstances& instances,
                             const size_t row_idx)
     {
-        auto base_univariates = instances.row_to_univariates(row_idx);
+        auto base_univariates = instances.template row_to_univariates<skip_count>(row_idx);
         for (auto [extended_univariate, base_univariate] : zip_view(extended_univariates.get_all(), base_univariates)) {
-            extended_univariate = base_univariate.template extend_to<ExtendedUnivariate::LENGTH>();
+            extended_univariate = base_univariate.template extend_to<ExtendedUnivariate::LENGTH, skip_count>();
         }
     }
 
-    template <typename Parameters, size_t relation_idx = 0>
-    void accumulate_relation_univariates(TupleOfTuplesOfUnivariates& univariate_accumulators,
-                                         const ExtendedUnivariates& extended_univariates,
+    template <typename UnivariateAccumulatorsType,
+              typename ExtendedUnivariatesType,
+              typename Parameters,
+              size_t relation_idx = 0>
+        requires(std::is_same_v<UnivariateAccumulatorsType, OptimisedTupleOfTuplesOfUnivariates> &&
```

Comment:
this should probable part of one or several concepts to aid readability

Reply:
or separate into two functions

Reply:
We can't produce concepts here, because the types are templated